### PR TITLE
Fix login form

### PR DESCRIPTION
**Pull request by duncangrant**
Logout was not redirecting correctly.  This adds a header to the
login form to the 401 api response so that it can be a) ignored by the
cli and b) followed by the UI.

### CODE REVIEW

**Review by iuliana — APPROVED** [review]
lgtm

**Comment by algairim** [comment]
Looks ok

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/filter/BrooklynSecurityProviderFilterJersey.java [hunk 1]
```diff
@@ -71,6 +74,11 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
                     rin = Response.status(Status.UNAUTHORIZED).entity("Authentication is required").build();
                 }
             }
+            if (rin.getStatus()==Status.UNAUTHORIZED.getStatusCode() &&
+                    Strings.isNonBlank(mgmt.getConfig().getConfig(BrooklynSecurityProviderFilterJavax.LOGIN_FORM))) {
+                rin = Response.status(Status.UNAUTHORIZED).entity("Authentication is required").header(LOGIN_PAGE_HEADER, mgmt.getConfig().getConfig(BrooklynSecurityProviderFilterJavax.LOGIN_FORM)).build();
```
Comment (ahgittin) [inline rest/rest-resources/src/main/java/org/apache/brooklyn/rest/filter/BrooklynSecurityProviderFilterJersey.java:79]: maybe cleaner if we make `rin` the `ResponseBuilder` and `build()` it when we return?  though the builder API might not support that.

**Comment by ahgittin** [comment]
merging

**Pull request merged**